command/agent/consul/syncer_test.go

```diff
@@ -95,148 +110,157 @@ func TestCheckRegistration(t *testing.T) {
+	service2 := &structs.Service{Name: "foo2"}
+	services := map[ServiceKey]*structs.Service{
+		GenerateServiceKey(service1): service1,
+		GenerateServiceKey(service2): service2,
+	}
+	if err := cs.SetServices(serviceGroupName, services); err != nil {
+		t.Fatalf("error setting services: %v", err)
+	}
 	if err := cs.SyncServices(); err != nil {
-		t.Fatalf("err: %v", err)
+		t.Fatalf("error syncing services: %v", err)
 	}
-	defer cs.Shutdown()
 
-	//Update Service defn 1
-	newTags := []string{"tag3"}
-	task.Services[0].Tags = newTags
+	// Now update both services
+	service1 = &structs.Service{Name: "foo1", Tags: []string{"a", "z"}}
+	service2 = &structs.Service{Name: "foo2", PortLabel: ":8899"}
+	service3 := &structs.Service{Name: "foo3"}
+	services = map[ServiceKey]*structs.Service{
+		GenerateServiceKey(service1): service1,
+		GenerateServiceKey(service2): service2,
+		GenerateServiceKey(service3): service3,
+	}
+	if err := cs.SetServices(serviceGroupName, services); err != nil {
+		t.Fatalf("error setting services: %v", err)
+	}
 	if err := cs.SyncServices(); err != nil {
-		t.Fatalf("err: %v", err)
-	}
-	// Make sure all the services and checks are still present
-	// service1 := &structs.Service{Name: task.Name}
-	// service2 := &structs.Service{Name: task.Name}
-	//services := []*structs.Service{service1, service2}
-	//service1.ServiceID = fmt.Sprintf("%s-%s:%s/%s", cs.GenerateServiceID(serviceGroupName, service1), task.Name, allocID)
-	//service2.ServiceID = fmt.Sprintf("%s-%s:%s/%s", cs.GenerateServiceID(serviceGroupName, service2), task.Name, allocID)
-	// if err := servicesPresent(t, services, cs); err != nil {
-	// 	t.Fatalf("err : %v", err)
-	// }
-	// FIXME(sean@)
-	// if err := checksPresent(t, []string{check1.Hash(service1ID)}, cs); err != nil {
-	// 	t.Fatalf("err : %v", err)
-	// }
-
-	// check if service defn 1 has been updated
-	// consulServices, err := cs.client.Agent().Services()
-	// if err != nil {
-	// 	t.Fatalf("errL: %v", err)
-	// }
-	// srv, _ := consulServices[service1.ServiceID]
-	// if !reflect.DeepEqual(srv.Tags, newTags) {
-	// 	t.Fatalf("expected tags: %v, actual: %v", newTags, srv.Tags)
-	// }
-}
+		t.Fatalf("error syncing services: %v", err)
+	}
 
-// func servicesPresent(t *testing.T, configuredServices []*structs.Service, syncer *Syncer) error {
-// 	var mErr multierror.Error
-// 	// services, err := syncer.client.Agent().Services()
-// 	// if err != nil {
-// 	// 	t.Fatalf("err: %v", err)
-// 	// }
-
-// 	// for _, configuredService := range configuredServices {
-// 	// 	if _, ok := services[configuredService.ServiceID]; !ok {
-// 	// 		mErr.Errors = append(mErr.Errors, fmt.Errorf("service ID %q not synced", configuredService.ServiceID))
-// 	// 	}
-// 	// }
-// 	return mErr.ErrorOrNil()
-// }
-
-func checksPresent(t *testing.T, checkIDs []string, syncer *Syncer) error {
-	var mErr multierror.Error
-	checks, err := syncer.client.Agent().Checks()
+	agentServices, err := cs.queryAgentServices()
```

Comment:
I changed the test above to not rely on this method's filtering.

I'd like to keep the helper method here and elsewhere since doing it manually everywhere seems like it would only add needless noise.